# HG changeset patch
# User Avinoam Kalma <a.kalma@gmail.com>
# Date 1481493957 -7200
#      Mon Dec 12 00:05:57 2016 +0200
# Node ID 7dec20b876d6dab89921ab6540d0fe7d561e6ad9
# Parent  84a52be0cf53173f2a0576eb85c41d2e7ac1accd
Add documentation for shape "valid" in function "conv" (bug #48218)

*conv.m: Add documentation for shape "valid"

diff --git a/scripts/polynomial/conv.m b/scripts/polynomial/conv.m
--- a/scripts/polynomial/conv.m
+++ b/scripts/polynomial/conv.m
@@ -16,29 +16,34 @@
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {} {} conv (@var{a}, @var{b})
 ## @deftypefnx {} {} conv (@var{a}, @var{b}, @var{shape})
 ## Convolve two vectors @var{a} and @var{b}.
 ##
-## The output convolution is a vector with length equal to
-## @code{length (@var{a}) + length (@var{b}) - 1}.
 ## When @var{a} and @var{b} are the coefficient vectors of two polynomials, the
 ## convolution represents the coefficient vector of the product polynomial.
 ##
-## The optional @var{shape} argument may be
+## The size of the result is determined by the optional @var{shape} argument
+## which takes the following values
 ##
 ## @table @asis
 ## @item @var{shape} = @qcode{"full"}
 ## Return the full convolution.  (default)
+## The result is a vector with length equal to
+## @code{length (@var{a}) + length (@var{b}) - 1}.
 ##
 ## @item @var{shape} = @qcode{"same"}
 ## Return the central part of the convolution with the same size as @var{a}.
+##
+## @item @var{shape} = @qcode{"valid"}
+## Return only the parts which do not include zero-padded edges.
+## The size of the result is @code{max (size (@var{a}) - size (@var{b}) + 1, 0)}.
 ## @end table
 ##
 ## @seealso{deconv, conv2, convn, fftconv}
 ## @end deftypefn
 
 ## Author: Tony Richardson <arichard@stark.cc.oh.us>
 ## Created: June 1994
 ## Adapted-By: jwe
